lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -3834,7 +3874,14 @@ class TFPartition : public SILFunctionTransform {
     if (!tfc.shouldBePartitioned(hostFn))
       return;
 
-    TFFunctionPartition partitioner(*hostFn, PM, *tfModule);
+    // If the function has @convention(tensorflow) representation, it is
+    // guaranteed to be an accelerator-only function. Send/receive is not
+    // allowed there.
+    bool isTFConvention =
```

Comment:
this should integrate with the patch @mhong sent out this morning.

Reply:
I'm fine with this code chunk in the context of this PR. we can discuss more details (e.g. whether to pass in isTFConvention or let partitioner recompute that via some method call over *hostFn) and the next-steps of frontend/backend change integration in person tomorrow.

Reply:
Yeah, I think having a method to check `hostFn.getRepresentation()` is better.